builder/dockerfile/parser/parser.go

```diff
@@ -176,6 +176,10 @@ func Parse(rwc io.Reader, d *Directive) (*Node, error) {
 				newline := scanner.Text()
 				currentLine++
 
+				// If escape followed by an empty line then stop
+				if strings.TrimSpace(newline) == "" {
```

Comment:
even though speed isn't a huge issue since I doubt we're parsing 1000's of lines, can we save the result of `strings.TrimSpace(newLine)` and re-use it above and below so we don't call it twice?